### PR TITLE
Add possibility to colors generation outside of widget

### PR DESCRIPTION
**Pull request by nikgalushko**
Some example usage:
```javascript
const generatedColors = DevExpress.viz.getPalette("Soft").generateColors(10);
const palette = DevExpress.viz.getPalette("Soft", { count: 17 });
const generateColors_2 = palette.generateColors();
```

### CODE REVIEW

File: js/viz/palette.js [hunk 1]
```diff
@@ -141,8 +151,8 @@ export function getPalette(palette, parameters) {
             result = palettes[currentPalette()];
         }
     }
-    result = result || null;
-    return type ? result ? result[type].slice(0) : result : result;
+
+    return type ? result[type].slice(0) : result;
```
Comment (pavelgruba) [inline js/viz/palette.js:155]: Is there a chance that results in undefined?

**Review by pavelgruba — CHANGES_REQUESTED** [review]
Re-format modules_metadata.json file to contain no extra changes.
Add new palette static methods to the ts/dx.all.d.ts

**Comment by roman-simionov** [comment]
> Add new palette static methods to the ts/dx.all.d.ts

That won't be necessary, so long as ts is generated by an internal tool from the documentation. So, the new palette methods should be described in doc files. it would be better to make a new type for the Pallete.

File: js/viz/palette.js (hunk 1 above)
Comment (nikgalushko) [reply js/viz/palette.js:155]: The result always is object because `currentPalette` always return a palette.

**Pull request closed without merging**